### PR TITLE
Upgrade to ruby 2.7.6

### PR DESCRIPTION
**Pull request by Dome-GER**
Upgrade ruby and remove the `therubyracer` gem causing libv8 conflicts
on MacOs.

`therubyracer` is an `execjs` engine to run JS, e.g. when building
assets. Assets are precompiled locally and copied by the Capistrano
tasks. So asset precompilation can fall back to the installed NodeJS
version.

This also upgrades the `rubocop` gem to not complain about ruby version 
mismatches and drops a seemingly outdated rake task.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,12 +19,17 @@ jobs:
     steps:
 
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+
+      - name: Setup NodeJS
```
Comment (Dome-GER) [inline .github/workflows/ci.yml:24]: Oh, this is probably not needed as we are precompiling assets in the Capistrano deployment tasks.

**Comment by SanJSp** [comment]
@Dome-GER 
While deploying I encountered folliwng issue:

Tasks: TOP => deploy:migrate
(See full trace by running task with --trace)
The deploy has failed with an error: Exception while executing as deploy@172.16.65.86: Exception while executing as deploy@172.16.65.86: rake exit status: 1
rake stdout: Nothing written
rake stderr: rake aborted!
ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes.
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/execjs-2.7.0/lib/execjs/runtimes.rb:58:in `autodetect'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/execjs-2.7.0/lib/execjs.rb:5:in `<module:ExecJS>'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/execjs-2.7.0/lib/execjs.rb:4:in `<top (required)>'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.4.7/lib/active_support/dependencies.rb:332:in `require'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.4.7/lib/active_support/dependencies.rb:332:in `block in require'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.4.7/lib/active_support/dependencies.rb:299:in `load_dependency'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/activesupport-6.1.4.7/lib/active_support/dependencies.rb:332:in `require'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/uglifier-4.2.0/lib/uglifier.rb:5:in `<top (required)>'
/var/www/hpi-career/releases/20220521131934/config/application.rb:7:in `<top (required)>'
/var/www/hpi-career/releases/20220521131934/Rakefile:4:in `require'
/var/www/hpi-career/releases/20220521131934/Rakefile:4:in `<top (required)>'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/bin/ruby_executable_hooks:22:in `eval'
/var/www/hpi-career/shared/bundle/ruby/2.7.0/bin/ruby_executable_hooks:22:in `<main>'

Does thi smean I simply have to install NodeJs on the staging server as recommended [here](https://stackoverflow.com/questions/8059332/rails-could-not-find-a-javascript-runtime-see-https-github-com-sstephenson-e/9655937#9655937)

**Comment by Dome-GER** [comment]
Yes, if NodeJS is installed, this should work.

**Comment by SanJSp** [comment]
As we do not have access to the JS config we maybe should try to find a solution for this without dropping the JS gem? I would revert this to finally finish the ToS Checkbox by testing it on staging

**Comment by Dome-GER** [comment]
I can check [mini_racer](`https://github.com/rubyjs/mini_racer) when there is some time.

**Pull request merged**